cmd/nknc/commands/asset.go

```diff
@@ -0,0 +1,195 @@
+	assetCmd.Flags().Uint32Var(&precision, "precision", 0, "asset precision")
+
+	assetCmd.Flags().MarkHidden("password")
+}
+
+func parseAddress() common.Uint160 {
+	if address := to; address != "" {
+		pg, err := common.ToScriptHash(address)
+		if err != nil {
+			fmt.Println("invalid receiver address")
+			os.Exit(1)
+		}
+		return pg
+	}
+	fmt.Println("missing flag [--to]")
+	os.Exit(1)
+	return common.EmptyUint160
+}
+
+func assetAction() error {
+	var err error
+	var txnFee common.Fixed64
+
+	if Fee == "" {
+		txnFee = common.Fixed64(0)
+	} else {
+		txnFee, err = common.StringToFixed64(Fee)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+	}
+
+	var resp []byte
+	switch {
+	case issue:
+		if walletFile == "" {
+			fmt.Println("wallet file is required with [--wallet]")
+			return nil
+		}
+
+		myWallet, err := vault.OpenWallet(walletFile, GetPassword(walletPassword))
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		if assetname == "" {
+			fmt.Println("asset name is required with [--name]")
+			return nil
+		}
+
+		if symbol == "" {
+			fmt.Println("asset symbol is required with [--symbol]")
+			return nil
+		}
+
+		totalSupply, err := common.StringToFixed64(value)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		if precision > config.MaxAssetPrecision {
+			err := fmt.Errorf("precision is larger than %v", config.MaxAssetPrecision)
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		txn, err := api.MakeIssueAssetTransaction(myWallet, assetname, symbol, totalSupply, precision, Nonce, txnFee)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		buff, err := txn.Marshal()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		resp, err = client.Call(Address(), "sendrawtransaction", 0, map[string]interface{}{"tx": hex.EncodeToString(buff)})
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+	case transfer:
+		myWallet, err := vault.OpenWallet(walletFile, GetPassword(walletPassword))
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		receipt := parseAddress()
+		amount, err := common.StringToFixed64(value)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		if Nonce == 0 {
+			remoteNonce, _, err := client.GetNonceByAddr(Address(), myWallet.Address, true)
+			if err != nil {
+				return err
+			}
+			Nonce = remoteNonce
+		}
+
+		txn, err := api.MakeTransferTransaction(myWallet, receipt, Nonce, amount, txnFee)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		buff, err := txn.Marshal()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		resp, err = client.Call(Address(), "sendrawtransaction", 0, map[string]interface{}{"tx": hex.EncodeToString(buff)})
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+	default:
```

Comment:
If no sub command is specified, the previous behavior is to print usage for this command, while the new behavior seems to be print nothing?

Reply:
yes, this is by design. if you look at L27 you will see that we use `RunE` and check for subcommands. if none present, we print usage, by the help of `showusage` boolean:

```
RunE: func(cmd *cobra.Command, args []string) error {
		showusage := true
		cmd.Flags().Visit(func(name *pflag.Flag) {
			showusage = false
		})
		if showusage {
			return cmd.Usage()
		}
		if err := assetAction(); err != nil {
			log.Error(err)
		}
		return nil
	},
```

this applies to all commands.

Reply:
To restore the previous behavior, we probably need to print asset cmd usage in this default branch. Same for all other commands default branch.

Reply:
> To restore the previous behavior, we probably need to print asset cmd usage in this default branch. Same for all other commands default branch.

yes. thats what I did. couldnt finish as I was busy lately. changes will come soon.